### PR TITLE
Correctly escape `'${`

### PR DESCRIPTION
**Pull request by Gabriel439**
Fixes https://github.com/dhall-lang/dhall-haskell/issues/2078

Previously the pretty-printer would escape `'${` as `'''${` inside of
a multi-line string, but that escape sequence is incorrect because
`'''` is itself an escape sequence for `''`.  The solution is to
instead escape `'${` as `${"'"}${`

### CODE REVIEW

File: dhall/src/Dhall/Pretty/Internal.hs [hunk 1]
```diff
@@ -1656,12 +1688,14 @@ consolidateRecordLiteral = concatMap adapt . Map.toList
...
+escapeSingleQuotedText :: Chunks s a -> Chunks s a
+escapeSingleQuotedText = splitWith f
   where
-    outputText = substitute "${" "''${" (substitute "''" "'''" inputText)
+    f inputText = Chunks [] outputText
+      where
+        outputText = substitute "${" "''${" (substitute "''" "'''" inputText)
 
-    substitute before after = Text.intercalate after . Text.splitOn before
+        substitute before after = Text.intercalate after . Text.splitOn before
```
Comment (mmhat) [inline dhall/src/Dhall/Pretty/Internal.hs:1698]: Why don't we use `Data.Text.replace` here?

**Comment by sjakobi** [comment]
> Shouldn't we also escape the `${` part of `'${`? If so, it should be `${"'"}''${`, right?

Indeed. I think that's just a little mistake in the description though. The test output looks correct, I think.

**Comment by basile-henry** [comment]
Shouldn't we also escape the `${` part of `'${`? If so, it should be `${"'"}''${`, right?

**Comment by basile-henry** [comment]
Oh yes you're right! I did look at the test, but I guess I didn't parse it correctly. Sorry about that :sweat_smile:

**Comment by sjakobi** [comment]
No worries, @basile-henry. I was initially surprised too! :)

**Pull request merged**